src/Tools/DynamoInstallDetective/Utilities.cs

```diff
@@ -30,6 +30,26 @@ public static IEnumerable FindDynamoInstallations(string additionalDynamoPath)
         }
 
         /// <summary>
+        /// Finds all unique Dynamo installation on the system that has file 
+        /// identifiable by the given fileLocator.
+        /// </summary>
+        /// <param name="additionalDynamoPath">Additional path for Dynamo binaries
+        /// to be included in search</param>
+        /// <param name="fileLocator">A callback method to locate dynamo specific files.</param>
+        /// <returns>List of KeyValuePair of install location and version info 
+        /// as Tuple. The returned list is sorted based on version info.</returns>
+        public static IEnumerable LocateDynamoInstallations(string additionalDynamoPath, Func<string, string> fileLocator)
```

Comment:
The additional path is for local built Dynamo which is not installed, just to support debugging seamlessly.